test/extended/builds/forcepull.go

```diff
@@ -1,183 +1,179 @@
-	}
-	o.Expect(len(hexIDs)).To(o.Equal(2))
-	resetData = map[string]string{s2iDockBldr: hexIDs[0], custBldr: hexIDs[1]}
-	g.By(fmt.Sprintf("\n%s FORCE PULL TEST:  hex id for s2i/docker %s and for custom %s ", time.Now().Format(time.RFC850), hexIDs[0], hexIDs[1]))
-})
+Also, in order to build the test case specific builder images only once, we currently have to do all the testing within a single g.It block.
+The project/namespace were being destroyed between tests, and that includes removal of the specific builder images
+we built.  The credentials also are recycled between those points.
 
-// TODO this seems like a weird restriction with segregated namespaces.  provide a better explanation of why this doesn't work
-// we don't run in parallel with this suite - do not want different tests tagging the same image in different ways at the same time
-var _ = g.Describe("builds: serial: ForcePull from OpenShift induced builds (vs. sti)", func() {
-	defer g.GinkgoRecover()
-	var oc = exutil.NewCLI("force-pull-s2i", exutil.KubeConfigPath())
+Dumping of the ImageStreams and Secrets at the various points proved this out.
+*/
 
-	g.JustBeforeEach(func() {
-		g.By("waiting for builder service account")
-		err := exutil.WaitForBuilderAccount(oc.AdminKubeREST().ServiceAccounts(oc.Namespace()))
-		o.Expect(err).NotTo(o.HaveOccurred())
-	})
+var _ = g.Describe("builds: forcepull: ForcePull from OpenShift induced builds (vs. sti)", func() {
+	defer g.GinkgoRecover()
+	var oc = exutil.NewCLI("forcepull", exutil.KubeConfigPath())
 
-	g.Describe("\n FORCE PULL TEST:  Force pull and s2i builder", func() {
-		// corrupt the s2i builder image
-		g.BeforeEach(func() {
-			exutil.CorruptImage(s2iDockBldr, custBldr, "s21")
-		})
+	g.BeforeEach(func() {
 
-		g.AfterEach(func() {
-			exutil.ResetImage(resetData)
-		})
+		g.By("refresh custom builder, prep forcepull builders")
+		exutil.PullImage(corruptor, dockerClient.AuthConfiguration{})
 
-		g.Context("\n FORCE PULL TEST:  when s2i force pull is false and the image is bad", func() {
+		// create the image streams and build configs for a test case specific builders
+		setupPath := exutil.FixturePath("fixtures", "forcepull-setup.json")
+		err := exutil.CreateResource(setupPath, oc)
 
-			g.It("\n FORCE PULL TEST s2i false", func() {
-				fpFalseS2I := exutil.FixturePath("fixtures", "forcepull-false-s2i.json")
-				g.By(fmt.Sprintf("\n%s FORCE PULL TEST s2i false:  calling create on %s", time.Now().Format(time.RFC850), fpFalseS2I))
-				exutil.StartBuild(fpFalseS2I, buildPrefix, oc)
+		// kick off the build for the new builder image just for force pull so we can corrupt them without conflicting with
+		// any other tests potentially running in parallel
+		exutil.StartBuild(s2iDockBldrPrefix, oc)
+		exutil.WaitForBuild("s2idock bldr build:  ", s2iDockBldrPrefix+"-1", oc)
+		exutil.StartBuild(custBldrPrefix, oc)
+		exutil.WaitForBuild("cust bldr build:  ", custBldrPrefix+"-1", oc)
```

Comment:
given that all your tests case about is "was this image pulled during the build" (and not whether the build actually succeeds/image was a valid builder) do you really need to create two distinct images here?  wouldn't one be sufficient?  you're already reusing one of them twice.